### PR TITLE
Conform to Kubernetes repo requirements

### PR DESCRIPTION
**Pull request by justaugustus**
Kubernetes repo requirements can be reviewed [here](https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md)
* Add CONTRIBUTING.md
* Add OWNERS
* Add OWNERS_ALIASES
* Add RELEASE.md
* Add SECURITY_CONTACTS
* Add code-of-conduct.md

Signed-off-by: Stephen Augustus <foo@agst.us>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Mentorship
+
+- [Mentoring Initiatives](https://git.k8s.io/community/mentoring) - We have a diverse set of mentorship programs available that are always looking for volunteers!
+
+<!---
+Custom Information - if you're copying this template for the first time you can add custom content here, for example:
+
+## Contact Information
+
+- [Slack channel](https://kubernetes.slack.com/messages/kubernetes-users) - Replace `kubernetes-users` with your slack channel string, this will send users directly to your channel.
```
Comment (tariq1890) [inline CONTRIBUTING.md]: Can we add uncomment and add the slack channel information here?

Comment (tariq1890) [reply CONTRIBUTING.md]:
```suggestion
 - [Slack channel](https://kubernetes.slack.com/messages/cluster-api-azure)
```

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Release Process
+
+The Kubernetes Template Project is released on an as-needed basis. The process is as follows:
```
Comment (tariq1890) [inline RELEASE.md]: I am assuming `Kubernetes Template Project`  is a placeholder?

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+1. An issue is proposing a new release with a changelog since the last release
+1. All [OWNERS](OWNERS) must LGTM this release
+1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
+1. The release issue is closed
+1. An announcement email is sent to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] kubernetes-template-project $VERSION is released`
```
Comment (tariq1890) [inline RELEASE.md]: same as above

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Release Process
+
+The Cluster API for Azure is released on an as-needed basis. The process is as follows:
+
+1. An issue is proposing a new release with a changelog since the last release
+1. All [OWNERS](OWNERS) must LGTM this release
+1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
+1. The release issue is closed
+1. An announcement email is sent to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] cluster-api-provider-azure $VERSION is released`
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply RELEASE.md]: SIG Azure will be the primary owner, with Cluster Lifecycle as a co-owner.

**Comment by justaugustus** [comment]
Merging this as it's been approved by multiple people.

**Pull request merged**